step.yml

```diff
@@ -59,27 +59,62 @@ toolkit:
     package_name: github.com/bitrise-steplib/steps-deploy-to-itunesconnect-application-loader
 
 inputs:
+- connection: "automatic"
+  opts:
+    title: Bitrise Apple Developer Connection
+    summary: The Apple Service authentication method the Step uses. By default, any enabled Bitrise Apple Developer connection is used and other authentication-related Step inputs are ignored.
+    description: |-
+      The input determines the method used for Apple Service authentication. By default, any enabled Bitrise Apple Developer connection is used and other authentication-related Step inputs are ignored.
+      
+      There are two types of Apple Developer connection you can enable on Bitrise: one is based on an API key of the App Store Connect API, the other is the legacy method of session-based authentication with an Apple ID. You can choose which type of Bitrise Apple Developer connection to use or you can tell the Step to only use Step inputs for authentication:
```

Comment:
do we want to mention the legacy, session-based wording? Or something like this: "..the other is the Apple ID authentication." :)